sprint 15

KonOsipov opened the pull request:
Спринт 15

KonOsipov committed:
sprint 15

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/data/SavedSettingsClient.kt:
```diff
@@ -0,0 +1,9 @@
+
+interface SavedSettingsClient {
+    fun getSaved(): SharedPrefsSettings
+    fun change()
```
Стоит изменить сигнатуру этой функции, т.к. функция с неймингом `change()` не понятно, что должна делать. Стоит дать ей более выразительное название в зависимости от логики, которая выполняется в ней

KonOsipov replied:
Спасибо, поправлю.

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/data/sharedPref/SharedPrefsManager.kt:
```diff
@@ -0,0 +1,61 @@
+
+            savedTracks.add(0, dto)
+
+            if (savedTracks.size > 10) savedTracks.removeAt(10)
```
Число 10 рекомендуется вынести в константу с говорящим названием для повышения читабельности кода

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/ui/tracklist/SearchActivity.kt:
```diff
@@ -20,239 +19,237 @@ import android.widget.LinearLayout
-                                progressBar.visibility = View.GONE
-                                if (response.body()?.results?.isNotEmpty() == true) {
+                trackList.clear()
+                trackInteractor = provideTracksInteractor(sharedPreferences)
```
Здесь не стоит при каждом поисковом запросе создавать новый интерактор. Рекомендуется создавать только 1 сущность интерактора при создании активити, чтобы уменьшить количество создаваемых объектов

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/ui/tracklist/SearchActivity.kt:
```diff
@@ -20,239 +19,237 @@ import android.widget.LinearLayout
 
-        inputEditText.setText(countValue)
 
         fun showSavedTracks() {
```
Здесь очень много функций объявлено внутри другой функции `onCreate`. Котлин хоть и позволяет объявлять функцию внутри другой, однако такие объявления сильно затрудняют чтение кода, смешивая объявления функций с их реальным использованием. Поэтому рекомендую объявления функций перетащить на уровень класса

ArturNurtdinov reviewed (commented):
1. Кнопка "Очистить" в поле ввода на экране поиска не видна:
![image](https://github.com/user-attachments/assets/8134567f-9575-48b7-97e9-fd27dbd15b2f)

KonOsipov committed:
sprint 15.1
1."-Кнопка "Очистить" в поле ввода на экране поиска не видна, клик не работает"
Поправил
2."Логика для работы с плеером не вынесена в интерактор, медиа плеер напрямую используется в активити. Нужно создать интерактор и репозиторий для работы с плеером"
Переделал PlayerActivity
